text/069-coq-roadmap.md

```diff
@@ -0,0 +1,316 @@
+- Title: Coq roadmap
+
+- Drivers: Nicolas Tabareau (@tabareau), Théo Zimmermann (@Zimmi48)
+
+----
+
+# Summary
+
+This CEP aims to establish a roadmap for the Coq project.
+It highlights both:
+
+- what are considered as important axes to work on in the future,
+- what resources are available to work on these axes, and which axes we commit
+  to work on based on these resources.
+
+Resources mean availability of persons to conduct the work. For technical axes,
+requiring changes in the Coq system, it includes availability of persons to
+review and merge the changes.
+
+# Motivation
+
+Producing a roadmap for the Coq project is important for several reasons:
+
+- It helps the community to know what to expect from the Coq project in the
+  future, in particular what changes can reasonably be expected to become
+  part of next releases, and what changes are more likely to be delayed
+  because of lack of resources.
+
+- It helps Coq developers to focus on important axes, and to be more
+  efficient, by making sure that their work will be supported by other
+  developers, including reviewers.
+
+- It helps contributors to know what are the important axes where their
+  contribution is most likely to be welcome and where they can expect
+  the most support from the Coq developers.
+
```

Comment:
I expect most contributors are motivated by their own ideas for improvements to the Coq and its ecosystem rather than asking "what project in this list do I want to join?".  This document is only a limited subset of the potential areas (axes?) for improvement.

> what are the important axes where their contribution is most likely to be welcome

Implying that some contributions are unwelcome is a poor choice of words.  I think the issue is more about getting reviewer time.  We should be willing to consider thoughtfully updating the roadmap at any time.  We should make a reasonable effort to review contributions that arrive unexpectedly--discussing their value rather than just pointing to the roadmap and saying "It's not in the roadmap.  Sorry but your contribution is unwelcome".

Reply:
> I expect most contributors are motivated by their own ideas for improvements to the Coq and its ecosystem rather than asking "what project in this list do I want to join?". This document is only a limited subset of the potential areas (axes?) for improvement.

If contributors are motivated by their own ideas for improvements, that's fine (although they should check with the maintainers if their ideas are likely to be accepted, when these ideas require large implementation work). However, there are also lots of contributors who are eager to help on priority axes (where they also know that they will receive the most help / feedback).

> This document is only a limited subset of the potential areas (axes?) for improvement.

We should strive to not limit the document in terms of general areas of improvement. However, we should limit the document to just a few priority items by area, because having too many priorities is just like having no priority.

> Implying that some contributions are unwelcome is a poor choice of words. I think the issue is more about getting reviewer time.

Yes. Better wording welcome!

> We should be willing to consider thoughtfully updating the roadmap at any time.

Yes. We discussed this at a recent Coq Call. We said that probably the roadmap would be copied to the wiki and updated there. We should make sure to add relevant pointers here when this is the case.

>We should make a reasonable effort to review contributions that arrive unexpectedly--discussing their value rather than just pointing to the roadmap and saying "It's not in the roadmap. Sorry but your contribution is unwelcome".

Yes. We should review all contributions that we receive. However, we should also make every effort to clearly communicate to potential contributors that large implementation works should not be started before the overall goal and design having been discussed with the maintainer (in an issue or a CEP).

Reply:
> However, we should also make every effort to clearly communicate to potential contributors that large implementation works should not be started before the overall goal and design having been discussed with the maintainer (in an issue or a CEP).

Yeah, but I'd phrase this more like "We encourage potential contributors to discuss their large planned contributions before they begin implementation, such as through a Github issue or a CEP."

Reply:
> > However, we should also make every effort to clearly communicate to potential contributors that large implementation works should not be started before the overall goal and design having been discussed with the maintainer (in an issue or a CEP).
> 
> Yeah, but I'd phrase this more like "We encourage potential contributors to discuss their large planned contributions before they begin implementation, such as through a Github issue or a CEP."

Yes, although I would probably use a word that has a similar meaning to "encourage", but stronger.

Reply:
"we urge" seems appropriate in this case.

Reply:
> "we urge" seems appropriate in this case.

Thanks, that's what I was looking for.